joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala

```diff
@@ -443,12 +443,31 @@ class DefaultTypeInfoProvider(environment: KotlinCoreEnvironment) extends TypeIn
         val renderedReturnType =
           if (renderedFqNameForDesc.startsWith(TypeConstants.kotlinApplyPrefix)) TypeConstants.javaLangObject
           else TypeRenderer.render(originalDesc.getReturnType)
-        val signature = s"$renderedReturnType($renderedParameterTypes)"
-        (s"$renderedFqName:$signature", signature)
+
+        val singleLambdaArgExprMaybe = expr.getSelectorExpression match {
+          case c: KtCallExpression if c.getLambdaArguments.size() == 1 =>
+            Some(c.getLambdaArguments.get(0).getLambdaExpression)
+          case _ => None
+        }
+        val fullNameSignature = s"$renderedReturnType($renderedParameterTypes)"
+        val signature =
+          singleLambdaArgExprMaybe
+            .map(lambdaInvocationSignature(_))
+            .getOrElse(fullNameSignature)
+        (s"$renderedFqName:$fullNameSignature", signature)
       case None => defaultValue
     }
   }
 
+  def lambdaInvocationSignature(expr: KtLambdaExpression): String = {
+    val paramsString =
+      if (expr.getValueParameters.isEmpty) TypeConstants.javaLangObject
+      else
+        TypeConstants.javaLangObject + s"${("," + TypeConstants.javaLangObject).repeat(expr.getValueParameters.size())}"
```

Comment:
Mixing string concatenation and interpolation looks a bit strange here. In my opinion, something like
`List.fill(expr.getValueParameters.size)(TypeConstants.javaLangObject).mkString(",") is cleaner, although that's very subjective.

Reply:
:+1: changed